rosserial_server/src/msg_lookup.cpp

```diff
@@ -35,6 +35,14 @@ namespace rosserial_server
 
 const MsgInfo lookupMessage(const std::string& message_type, const std::string submodule)
 {
+  // Lazy-initialize the embedded Python interpreter. The total
+  // memory cost of having this in process is only about 5-6MB.
+  static bool py_initialized = false;
+  if (!py_initialized)
```

Comment:
This check is not really needed. Calling `Py_Initialize()` a second time is considered a no-op, if we really do want to check, we could use `Py_IsInitialized()`.
```suggestion
  if (!Py_IsInitialized())
```

https://docs.python.org/3.5/c-api/init.html#c.Py_IsInitialized

This could possibly be still messed up if `Py_Finalize()` is called somewhere else.

Reply:
> Calling `Py_Initialize()` a second time is considered a no-op

Is this the case? I'm unable to find documentation supporting this.